src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -256,6 +258,24 @@ private Map<Integer, Float> doANNSearch(
             );
         }
 
+        QuantizationParams quantizationParams = quantizationService.getQuantizationParams(fieldInfo);
+
+        byte[] quantizedVector = null;
+
+        if (quantizationParams != null) {
+            QuantizationConfigKNNCollector tempCollector = new QuantizationConfigKNNCollector();
+            reader.searchNearestVectors(knnQuery.getField(), new float[0], tempCollector, null);
+            if (tempCollector.getQuantizationState() == null) {
+                throw new IllegalStateException("No quantization state for file");
+            }
+            QuantizationOutput quantizationOutput = quantizationService.createQuantizationOutput(quantizationParams);
+            quantizedVector = (byte[]) quantizationService.quantize(
```

Comment:
Please add a TODO on top of this line which suggest that in future it is not necessary that we will have byte[] as an output from quantize function.